python/src/aac/lang/lsp/server.py

```diff
@@ -0,0 +1,84 @@
+"""The Architecture-as-Code Language Server."""
+
+import logging
+from typing import Optional
+from pygls.server import LanguageServer
+import pygls.lsp.methods as methods
+
+from pygls.lsp import (
+    CompletionOptions,
+    CompletionParams,
+    Hover,
+    HoverParams,
+    InitializeParams,
+    InitializeResult,
+    ServerCapabilities,
+)
+
+from aac.lang.active_context_lifecycle_manager import get_active_context
+from aac.lang.language_context import LanguageContext
+from aac.lang.lsp.code_completion_provider import CodeCompletionProvider
+
+
+LOGGER: Optional[logging.Logger] = None
+SERVER: Optional[LanguageServer] = None
+ACTIVE_CONTEXT: Optional[LanguageContext] = None
+COMPLETION_PROVIDER: Optional[CodeCompletionProvider] = None
+
+
+def start_lsp(dev: bool = False):
+    """Start the LSP server.
+
+    Args:
+        dev (bool): Whether to start the server and setup logging for development. (optional)
+    """
+    global SERVER, LOGGER, ACTIVE_CONTEXT
+
+    SERVER = SERVER or LanguageServer()
+    setup_features(SERVER)
+    ACTIVE_CONTEXT = get_active_context()
```

Comment:
Is there a reason to make this a global? It seems like we can make it local and just use `get_active_context` when we want to access the active context.